Yank/restraints.py

```diff
@@ -1476,7 +1516,9 @@ def determine_missing_parameters(self, thermodynamic_state, sampler_state, topog
                              'restraint parameters...'.format(attempt, MAX_ATTEMPTS))
 
                 # Randomly pick non-collinear atoms.
-                self.restrained_atoms = self._pick_restrained_atoms(sampler_state, topography)
+                restrained_atoms = self._pick_restrained_atoms(sampler_state, topography)
+                self.restrained_receptor_atoms = restrained_atoms[:3]
```

Comment:
Do we want this to auto-repick both the receptor AND ligand atoms? What if the user specifies a series of receptor atoms they always want to use and dont care about the ligand atoms? This method will re-pick the receptor ones as well.